### ISSUE TITLE
Setting background image to color with roundAsCircle="true" should be masked by circle

### ISSUE THREAD

**Issue by emhagman** [facebook/fresco#57]
At the moment, displaying the image with `roundAsCircle` set to `true` creates a square around the circle as a background when setting `backgroundImage` to `@color/white` or any other color resource.

In my opinion, it makes sense for the circle to be used as a mask for the background.

Current code:
<img src="https://cloud.githubusercontent.com/assets/1421702/6934307/6346d008-d800-11e4-931d-ee18aad0def0.png" width="130" height="100" />

Proposed fix:
<img src="https://cloud.githubusercontent.com/assets/1421702/6934310/6529630e-d800-11e4-9f24-ec93e398de88.png" width="130" height="100" />

Appologies for the teribble photo shopping, just trying get point across.

**Comment by blackiedm** [comment]
in this case, you can use RoundingParams.RoundingMethod.BITMAP_ONLY:
just like this:
```
RoundingParams roundingParams = new RoundingParams();
roundingParams.setRoundAsCircle(true);
roundingParams.setBorder(R.color.border_color, 10);
roundingParams.setRoundingMethod(RoundingParams.RoundingMethod.BITMAP_ONLY);
```
and set it to `GenericDraweeHierarchyBuilder`;

**Comment by plamenko** [comment]
The thing is that we do not apply rounding on overlays and backgrounds when using the `BITMAP_ONLY` mode (which is default). I'll try to send a fix for that.

But, we do provide `RoundedColorDrawable`, so as a workaround, you can use that if you are creating your hierarchy programmatically. 
https://github.com/facebook/fresco/blob/847851a8de6d223c0a6634447bddee1252e01c64/drawee/src/main/java/com/facebook/drawee/drawable/RoundedColorDrawable.java

**Comment by tyronen** [comment]
Fixed in 0.3.0.
